router/patients.js

```diff
@@ -1,36 +1,33 @@
 const express = require("express");
+const { getPatient, getPatientPrescriptions } = require("../services/patientServise");
 const router = express.Router();
 
-const User = require("../models/userModel");
 
 router.get("/:id", async (req, res) => {
   try {
     const patientId = req.params.id;
-    const user = await User.findOne({ _id: patientId, role: "patient" });
+    const user = await getPatient(patientId)
     if (!user) {
       throw new Error("Patient is not found");
     }
     res.status(200).json(user);
   } catch (error) {
-    res.status(404).json({ message: error });
+    const errorMessage = error.message || "Internal Server Error";
+    res.status(404).json({ message: errorMessage });
   }
 });
 
 router.get("/:id/prescriptions", async (req, res) => {
   try {
     const patientId = req.params.id;
-    const userPrescriptions = await User.findOne({
-      _id: patientId,
-      role: "patient",
-    })
-      .populate("prescriptions")
-      .select("prescriptions");
-    if (!user) {
-      throw new Error("Patient is not found");
+    const userPrescriptions = await getPatientPrescriptions(patientId)
```

Comment:
`userPrescriptions` could be renamed to `patientPrescriptions` or just `prescriptions` as it is in the patients route.